src/layer/x86/scale_x86.cpp

```diff
@@ -35,53 +35,92 @@ int Scale_x86::forward_inplace(std::vector<Mat>& bottom_top_blobs, const Option&
+    const float* scale = scale_blob;
+    const float* bias = bias_data;
+
+    if (dims == 1)
     {
-        if (dims == 1)
-        {
-            int w = bottom_top_blob.w;
+        float* ptr = (float*)bottom_top_blob;
+        int size = w * elempack;
 
-            const float* scale = scale_blob;
+#if __AVX__
+        int nn = size >> 3;
+        int remain = size & 7;
+        #pragma omp parallel for num_threads(opt.num_threads)
+        for (int i = 0; i < nn; i++)
+        {
+            __m256 _p = _mm256_loadu_ps(ptr);
+            __m256 _s = _mm256_loadu_ps(scale);
             if (bias_term)
             {
-                const float* bias = bias_data;
-                #pragma omp parallel for num_threads(opt.num_threads)
-                for (int i = 0; i < w; i++)
-                {
-                    float* ptr = (float*)bottom_top_blob + i * 8;
+                __m256 _bias = _mm256_loadu_ps(bias);
+                _p = _mm256_comp_fmadd_ps(_p, _s, _bias);
+                bias += 8;
+            }
+            else
+            {
+                _p = _mm256_mul_ps(_p, _s);
+            }
+            _mm256_storeu_ps(ptr, _p);
 
-                    __m256 _p = _mm256_loadu_ps(ptr);
-                    __m256 _s = _mm256_loadu_ps(scale + i * 8);
-                    __m256 _bias = _mm256_loadu_ps(bias + i * 8);
-                    _p = _mm256_comp_fmadd_ps(_p, _s, _bias);
-                    _mm256_storeu_ps(ptr, _p);
-                }
+            ptr += 8;
+            scale += 8;
+        }
+#else
+        int nn = size >> 2;
+        int remain = size & 3;
+        #pragma omp parallel for num_threads(opt.num_threads)
+        for (int i = 0; i < nn; i++)
+        {
+            __m128 _p = _mm_loadu_ps(ptr);
+            __m128 _s = _mm_loadu_ps(scale);
+            if (bias_term)
+            {
+                __m128 _bias = _mm_loadu_ps(bias);
+                _p = _mm_comp_fmadd_ps(_p, _s, _bias);
+                bias += 4;
             }
             else
             {
-                #pragma omp parallel for num_threads(opt.num_threads)
-                for (int i = 0; i < w; i++)
-                {
-                    float* ptr = (float*)bottom_top_blob + i * 8;
+                _p = _mm_mul_ps(_p, _s);
+            }
+            _mm_storeu_ps(ptr, _p);
 
-                    __m256 _p = _mm256_loadu_ps(ptr);
-                    __m256 _s = _mm256_loadu_ps(scale + i * 8);
-                    _p = _mm256_mul_ps(_p, _s);
-                    _mm256_storeu_ps(ptr, _p);
-                }
+            ptr += 4;
+            scale += 4;
+        }
+#endif // __AVX__
+
+        for (int i = 0; i < remain; i++)
+        {
+            if (bias_term)
+            {
+                *ptr = *ptr * *scale + *bias;
+                bias++;
+            }
+            else
+            {
+                *ptr = *ptr * *scale;
             }
+
+            ptr++, scale++;
         }
 
+        return 0;
+    }
```

Comment:
for dim==1 case, move bias condition out of the loop

data race happens in ptr and scale pointer with multi-threading